crates/core/src/client/mod.rs

```diff
@@ -606,4 +666,56 @@ impl<P: Provider + Send + Sync> KakarotStarknetApi<P> for KakarotClient<P> {
             Ok(starknet_block.to_eth_block(self).await)
         }
     }
+
+    /// Get the simulation of the BroadcastedInvokeTransactionV1 result
+    async fn simulate_transaction(
```

Comment:
let's create an issue to update simulate to make it agnostic to provider
add a TODO

Reply:
to make sure I understand correcly, we need to write this all because it's not yet usable in starknet-rs? will the RPC fails consistenly when using madara/katana because this will raise?

Reply:
Yes, if you try to use `simulate_transaction` in Katana or Madara, this will raise when trying to get the `gateway_url` on this line:
```
let mut url = self
            .network
            .gateway_url()?
            .join("simulate_transaction")
            .map_err(|e| EthApiError::FeederGatewayError(format!("gateway url parsing error: {:?}", e)))?;
```